swatch-contracts/src/main/java/com/redhat/swatch/contract/service/ContractService.java

```diff
@@ -230,41 +236,127 @@ public StatusResponse callPartnerApiAndUpsertPartnerContract(
   public ContractMessageProcessingResult upsertPartnerContract(
```

Comment:
I'd like to see more comments in this method.  It's doing a lot: deletions, updates, subscription updates.  If you wanted to break it up, that would be fine too, but it took me awhile to wrap my head around everything this Goliath is doing and how it's doing it.

Reply:
Ya agreed I'll probably break this up into a more readable methods.